crates/polars-arrow/src/legacy/kernels/rolling/nulls/variance.rs

```diff
@@ -178,10 +178,10 @@ impl<
 
         let denom = count - ddof;
 
-        if count == T::zero() {
+        if denom <= T::zero() {
             None
         } else if count == T::one() {
-            NumCast::from(0)
+            Some(T::zero())
```

Comment:
like this it matches what's done for the `no_nulls/variance.rs` kernel

https://github.com/pola-rs/polars/blob/b6187346238f5d72fbe2a5a529817dbadc1b7c32/crates/polars-arrow/src/legacy/kernels/rolling/no_nulls/variance.rs#L121-L125